common/scala/src/main/scala/org/apache/openwhisk/core/entity/Parameter.scala

```diff
@@ -224,9 +249,32 @@ protected[core] object Parameters extends ArgNormalizer[Parameters] {
   protected[core] def apply(p: String, v: JsValue): Parameters = {
     require(p != null && p.trim.nonEmpty, "key undefined")
     Parameters() + (new ParameterName(ArgNormalizer.trim(p)),
-    ParameterValue(Option(v).getOrElse(JsNull), false))
+    ParameterValue(Option(v).getOrElse(JsNull), false, None))
   }
 
+  def readMergedList(value: JsValue): Parameters =
+    Try {
+
+      val JsObject(obj) = value
+      new Parameters(
+        obj
+          .map((tuple: (String, JsValue)) => {
+            val key = new ParameterName(tuple._1)
+            val paramVal: ParameterValue = tuple._2 match {
+              case o: JsObject =>
+                o.getFields("value", "init", "encryption") match {
```

Comment:
I suspect there is some ambiguity here: if a user passes a parameter to the action that conforms to this schema, it could end up matching here.